### DISCUSSION TITLE
How cutlass control reading global memory when TileSize can not be divided by ProblemSize?

### DISCUSSION THREAD

**Discussion by MARD1NO** [NVIDIA/cutlass#1507] [Q&A] [answered]
Assume ProblemSize M N K is 63 63 63, and the TileSize is 64 64 64, how cutlass control the reading of global memory to prevent ilegal memory access? I'm not very clear about the details. (Maybe the key is predication in cp_async?)

**Comment by hwu36** [comment, marked as answer]
yes, prediction will tell you if the loading address is out of bound and then cp.async will not be executed.

the other key thing is that you need to set the alignment to be 1 so that every thread loads one data a time and the loading address can be any -- i.e. does not have to be aligned with multiple elements.
